thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -140,21 +140,49 @@ class CDistributorBase : implements IHashDistributor, implements IExceptionHandl
              */
             dedupList.sort(*iCompare, 1);
 
-            OwnedConstThorRow prev;
-            for (unsigned i = c; i>0;)
+            if (!keepBestCompare)
             {
-                OwnedConstThorRow row = dedupList.getClear(--i);
-                if ((NULL != prev.get()) && (0 == iCompare->docompare(prev, row)))
+                OwnedConstThorRow prev;
+                for (unsigned i = c; i>0;)
                 {
-                    /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
-                     * which will be deducted when sent.
-                     */
+                    OwnedConstThorRow row = dedupList.getClear(--i);
+                    if ((NULL != prev.get()) && (0 == iCompare->docompare(prev, row)))
+                    {
+                        /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
+                        * which will be deducted when sent.
+                        */
+                    }
+                    else
+                    {
+                        prev.set(row);
+                        rows.enqueue(row.getClear());
```

Comment:
I was mistaken, the Best block below, will dedup as expect, and only serialize the best candidates.